cmd/zpool/zpool_main.c

```diff
@@ -6556,6 +6561,46 @@ status_callback(zpool_handle_t *zhp, void *data)
 	return (0);
 }
 
+/*
+ * Dumps pool's config in json
+ */
+int
+dump_callback(zpool_handle_t *zhp, void *data)
+{
+	nvlist_t *config;
+	char *buf;
+
+	config = zpool_get_config(zhp, NULL);
+
+	if (config != NULL) {
+		nvlist_dump_json(config, &buf);
+		printf("%s\n", buf);
+		nvlist_dump_json_free(config, buf);
+	}
+	return (0);
+}
+
+/*
+ * zpool dump
+ *
+ * Dumps entire config in json format
+ */
+int
+zpool_do_dump(int argc, char **argv)
+{
+	int ret;
+	status_cbdata_t cb = { 0 };
+	cb.cb_allpools = B_TRUE;
```

Comment:
both the lines are not needed.

Reply:
we need to set cb_allpools to iterator to do callback for all pools